src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseTestService.java

```diff
@@ -607,13 +607,11 @@ void createProgrammingExerciseForExam_invalidExercise_dates(InvalidExamExerciseD
     // TEST
     void createProgrammingExerciseForExam_DatesSet() throws Exception {
         setupRepositoryMocks(examExercise, exerciseRepo, solutionRepo, testRepo, auxRepo);
-        ExerciseGroup exerciseGroup = examExercise.getExerciseGroup();
```

Comment:
Why is this removed?

Reply:
This is due to the chage I made to `generateProgrammingExerciseForExam`. This test is the only one that did not expect an exam programming exercise being added to the exercise group per default. I think if an exam programming exercise is created and we have `exercise.setExerciseGroup(exerciseGroup);` we should also set `exerciseGroup.addExercise(programmingExercise);`. The last assertion in the test was in my option not really needed, since a Bad Request already implies the post call failing and the exercise not being added to the exercise group